pkg/repo/util.go

```diff
@@ -17,39 +16,31 @@ import (
 
 // initRepoFiles initializes all files required for a valid bacalhau repo.
 func initRepoFiles(cfg types.BacalhauConfig) error {
```

Comment:
The repo remains responsible for initializing paths under it. The difference after this change is that a repo is _given_ a config, and it initializes itself based on the config values. I have kept the repo responsible for initializing paths for several reasons:
1. https://github.com/bacalhau-project/bacalhau/pull/4330#discussion_r1719012969
2. The config doesn't initialize paths because the config isn't guaranteed to have a repo path set, especially the case for testing. We are unable to create a default config with the right paths because such a config would require the correct repo path defined in it, and that's not possible.
3. This keeps the repo responsible for mutating the file system, and allows the config to fully dictate how the repo is initialized.

Reply:
Configs should NOT have an empty repo. Either a user provides a repo path, or we fall back to default one _and_ set it in the config, but we should have validation and fail if config ends with an empty repo

Reply:
see latest changes